### PR TITLE
Fix bug in basic_iterator operator--(int). Add test case for same.

### PR DESCRIPTION
**Pull request by Beman**
Added test case was failing before fix. OK after fix.

HTH,

--Beman

### CODE REVIEW

File: include/stl2/detail/iterator/basic_iterator.hpp [hunk 1]
```diff
@@ -722,7 +722,7 @@ STL2_OPEN_NAMESPACE {
     {
       auto tmp = *this;
       --*this;
-      return *this;
+      return tmp;
```
Comment (ericniebler) [inline include/stl2/detail/iterator/basic_iterator.hpp]: Or simply `return prev(*this);`

Comment (CaseyCarter) [inline include/stl2/detail/iterator/basic_iterator.hpp]: > Or simply `return prev(*this);`

We want the "old" value which is `next(*this)`, not `prev(*this)`. `--*this; return next(*this);` would be semantically equivalent, but I conjecture that `auto tmp = *this; --*this; return tmp;` is cheaper in general. A copy must be made in any case, copying the original value will be cheaper if the optimizer can't determine that `++` simply reverses the effect of `--`.

Comment (ericniebler) [inline include/stl2/detail/iterator/basic_iterator.hpp]: Sorry, looked

Comment (ericniebler) [inline include/stl2/detail/iterator/basic_iterator.hpp]: too quickly. :-P

**Pull request merged**